### PR TITLE
Fix peaktypeestimator

### PR DESCRIPTION
**Pull request by cbielow**
this PR makes the PeakTypeEstimator very robust. Its works on all data tested (Orbi, FTMS, IonTrap),
even on very sparse spectra (which have been very problematic in the past) and extreme m/z ranges (where the inter-peak spacing usually becomes very wide).

Accuracy is 100% on the BSA example data (centroided) in share, PeakTypeTest data and my 16k Orbitrap spectra test set (profile & centroided).

### CODE REVIEW

**Comment by timosachsenberg** [comment]
Cool. Did you test this on high-mass + high-charge isotope patterns (e.g., top-down?).

**Comment by timosachsenberg** [comment]
And you could also use: https://github.com/OpenMS/OpenMS/blob/develop/src/tests/topp/MetaProSIP_1_input.mzML
as test-case, if you want.
It contains a 50% 13C labeled peptide (unfortunately only charge 2 so I don't know if all 5 peaks fall into the 1 Da window)

**Comment by cbielow** [comment]
good idea. I did not think of heavily labeled peptides.

MetaProSIP dataset has 57 centroid spectra, correct .-)

I tried on a top down dataset from pride. Profile spectra were correctly identified 99% of the time.
Centroid spectra 100%. But the test dataset was not very extensive.

**Comment by hroest** [comment]
seems like there are 29 failing tests right now, can you have a look at these?

File: src/openms/include/OpenMS/FORMAT/PeakTypeEstimator.h [hunk 1]
```diff
@@ -106,8 +106,9 @@ namespace OpenMS
         if (idx == -1) break;
 
         // check left and right peak shoulders and count number of sample points
-        PeakConstIterator it_max = data.begin() + idx;
-        PeakConstIterator it = it_max;
+        typedef typename std::vector<PeakT>::iterator PeakIterator; // non-const version, since we need to modify the peaks
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/PeakTypeEstimator.h]: would:
using PeakIterator = std::vector<PeakT>::iterator; 
work?

Comment (timosachsenberg) [reply src/openms/include/OpenMS/FORMAT/PeakTypeEstimator.h]: Actually I meant with the brackets but seems this now also works ;)

**Pull request merged**